### PR TITLE
Add seven segment display

### PR DESCRIPTION
**Pull request by daviscook477**
That's all it does

### CODE REVIEW

File: ONI-DenseLogic/DenseLogicStrings.cs [hunk 1]
```diff
@@ -37,6 +37,19 @@ public static class DENSELOGICTEAM_DENSEGATE {
...
 				}
 
+				public static class DENSELOGICTEAM_LOGICSEVENSEGMENT {
+					public static LocString NAME = UI.FormatAsLink("Seven Segment Display",
+						LogicSevenSegmentConfig.ID);
+					public static LocString DESC = $"Giving the duplicants a way to show off their favorite numbers has never been easier.";
+					public static LocString EFFECT = $"Displays the value of a {UI.FormatAsLink(STRINGS.BUILDINGS.PREFABS.LOGICRIBBON.NAME, LogicRibbonConfig.ID)} as a number 0 through 9.";
+					public static LocString LOGIC_PORT = "Numerical Input";
+					public static LocString INPUT_PORT_ACTIVE = $"Display a non-zero number.";
+					public static LocString INPUT_PORT_INACTIVE = $"Display zero.";
```
Comment (Hexicube) [inline ONI-DenseLogic/DenseLogicStrings.cs]: INPUT_PORT_INACTIVE should be "Displays zero."

File: ONI-DenseLogic/LogicSevenSegment.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		public void Render200ms(float dt) {
+			// hexi/test/peter: Do we have to do this here? Can we render only on state change?
+			UpdateMeter();
+			UpdateVisuals();
+		}
+
+		public void UpdateVisuals() {
+			int num0 = inVal > 0 ? 1 : 0;
+			int num1 = curOut;
+			int state = num0 + 4 * num1;
```
Comment (Hexicube) [inline ONI-DenseLogic/LogicSevenSegment.cs:81]: Should this not be "4 * num0 + num1"?

Also, "inVal > 0 ? 1 : 0" could be merged into this statement as "inVal > 0 ? 4 + num1 : num1" or "1 + num1 * 4 : num1 * 4" depending on where the multiplication is supposed to go.

**Review by Hexicube — CHANGES_REQUESTED** [review]

File: ONI-DenseLogic/DenseLogicStrings.cs (hunk 1 above)
Comment (daviscook477) [reply ONI-DenseLogic/DenseLogicStrings.cs]: yes I agree with asquared. I'll change it.

File: ONI-DenseLogic/LogicSevenSegment.cs (hunk 2 above)
Comment (daviscook477) [reply ONI-DenseLogic/LogicSevenSegment.cs:81]: I did it this way to be consistent with the way I've written out the `on_{number}` calculations for the other things. I suppose I could change all of them but I think I'll leave this as it. I think it reads pretty clearly.

**Pull request merged**